x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java

```diff
@@ -381,6 +381,32 @@ public void testAutoExpandDuringReplace() throws Exception {
         assertIndexSetting("index", "index.number_of_replicas", "1");
     }
 
+    public void testAutoExpandDuringShutdown() throws Exception {
+
+        var node1 = internalCluster().startNode();
+        var node2 = internalCluster().startNode();
+
+        createIndex("index", indexSettings(1, 0).put("index.auto_expand_replicas", randomFrom("0-all", "0-1")).build());
+        indexRandomData("index");
+
+        ensureGreen("index");
+        assertIndexSetting("index", "index.number_of_replicas", "1");
+
+        var nodeNameToShutdown = randomFrom(node1, node2);
+        var nodeIdToShutdown = getNodeId(nodeNameToShutdown);
+
+        putNodeShutdown(nodeIdToShutdown, SingleNodeShutdownMetadata.Type.REMOVE, null);
```

Comment:
Should we choose randomly and include `SIGTERM` and `REPLACE`? (based on [this change](https://github.com/elastic/elasticsearch/pull/103251/files#diff-9e623e836db0187186ec9be9a61fa88d45b3e5499b6e9d2bb315aafa5955f43bR110))